docs/deployment-integration.md

```diff
@@ -0,0 +1,312 @@
+# Deployment of the Rollup BSN Contract
+
+## Introduction
+
+The Rollup BSN contract is a CosmWasm smart contract deployed on
+the Babylon Genesis chain that tracks finality signatures for rollup blocks.
+It verifies signatures, timestamps public randomness, detects equivocation, and
+reports misbehavior for slashing. This is the core and only
+contract that a rollup deploys to become a BSN. The contract must be deployed
+on Babylon Genesis as it relies on direct interaction with it.
+
+## Governance Notes
+<img width="3100" height="772" alt="governance" src="https://github.com/user-attachments/assets/f1170b6c-5666-4ae9-91cc-49067f97a3a1" />
+
+
+
+Depending on the Babylon Genesis network (e.g., testnet, mainnet) you
+choose for the deployment of the rollup BSN, the network might be permissioned:
+* **Permissioned CosmWasm**: Requires governane approval for deploying a smart
+  contract. This can be granted in two ways:
+  * `MsgStoreCode` proposal: Upload a contract code. Preferred for one-time deployments
+  * `MsgUpdateParams` proposal: Add a Babylon Genesis address in an allow-list
+    for which its members can permissionlessly upload code. This is typically
+    used for contracts requiring periodic maintenance.
+  > To learn more about permissioned CosmWasm,
+  > see the [deployment
+  > guide](https://docs.babylonlabs.io/guides/governance/submit_proposals/smart_contract_deployment/) 
+  > and review past proposals in the [governance
+  > forum](https://forum.babylon.foundation/c/gov-proposals/smart-contract-proposals/15).
+* **Permissioned BSN Registration**: Requires a governance approval for
+  the registration of a BSN. This can be granted via the submission of
+  a governance proposal executing a `MsgRegisterConsumer` using the
+  metadata defined in the [Rollup BSN Registration](#rollup-bsn-registration)
+  section.
+
+  For governance guidelines, please refer to the governance forum: 
+  [Babylon Governance Guidelines](https://forum.babylon.foundation/c/gov-proposals/9/none)
```

Comment:
Instead of having this text here, how about we add a TODO to fill this when there's a section for BSN registration props?